internal/policy/policy.go

```diff
@@ -170,10 +172,38 @@ func LoadStateForEntry(ctx context.Context, repo *git.Repository, e rsl.EntryTyp
 
 		if entry.Name == fmt.Sprintf("%s.json", RootRoleName) {
 			state.RootEnvelope = env
+			rootMetadata, err := state.GetRootMetadata()
+			if err != nil {
+				return nil, err
+			}
+			for _, key := range rootMetadata.Keys {
+				key := key // avoid iterator pointer issues
+				key.ID()   //nolint:errcheck
```

Comment:
@wlynch any thoughts on how we can ensure keyID is set when we're pulling tuf.Key out of the metadata outside of including KeyID in the serialized form?

Reply:
The issue was with how keys were being serialized in metadata. It was `map[keyID]: key` except `key` didn't have `.KeyID` set. So when reading in a key from metadata we'd either have to recalculate it or do some processing when looping through the map to use the key there. Either way, it was messy. In #61, I've just included the serializing of key IDs in the map. This actually aligns with in-toto as it happens but perhaps not some TUF implementations. Also, #61 offloads this to go-securesystemslib so we don't have to keep tweaking things here.